WPMedia/fusion-news-theme-blocks
# Adding recaptcha_v3

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

Render reCaptcha v3 in case the tenant has this config.

We are passing a parameter

reCaptcha V3 is the invisible reCaptcha provided by google. Once it's included on the page should look like the image below.
![Screenshot 2024-03-20 at 4 16 19 p m](https://github.com/WPMedia/arc-themes-blocks/assets/54566275/4ef0b963-5e18-4220-9ef3-7fa38470667e)


By default, google includes this in the bottom right corner.
reCaptcha v3 does not require action by the user, the recaptcha is returned automatically.

BotChallengeProtection component will accept a challengeIn attribute. This attribute could be one of:
'signup' || ‘signin’ || ‘magiclink’ || ‘checkout’

IF !!`${challengeIn}Recaptcha && (0 <=recaptchaScore && recaptchaScore <= 1) && !!recaptchaSiteKey
THEN render reCaptchaV3. reCaptchaV3 should be initialized by passing the recaptchaSiteKey

Currently, we have this enabled for org:subssinglesite site:the-daily-intelligencer env: sandbox [admin](https://sandbox.subssinglesite.arcpublishing.com/subscriptions/settings/security/recaptcha?site=the-daily-intelligencer). 
findeIn case you need this setting for any tenant, please let us know.

## Jira Ticket

[ASUB-8193](https://arcpublishing.atlassian.net/browse/ASUB-8193)

## Acceptance Criteria

Update BotChallengeProtection component in [identity-block](https://github.com/WPMedia/arc-themes-blocks/tree/arc-themes-release-version-2.1.4/blocks/identity-block)

BotChallengeProtection will accept a challengeIn attribute. This attribute could be one of:
'signup' || ‘signin’ || ‘magiclink’ || ‘checkout’

IF !!`${challengeIn}Recaptcha && recaptchaScore between 0 - 1 &&!!recaptchaSiteKey
THEN render reCaptchaV3. reCaptchaV3 should be initialized by passing the recaptchaSiteKey

Every time you grab a response token from reCAPTCHA component, store the response in localStorage.

The BotChallengeProtection should be part of the Login block, and the reCaptcha token should be sent during the call to Identity.login();
User should be successfully Logged In when reCaptcha v3 is enabled for signin.

## Test Steps

[Figma file](https://www.figma.com/file/NBjMF1v4mzwv9jkn7NEq9V/branch/DpQUSctnVjSgVsmUGT9smx/Themes-Block-Styling-2.0?type=design&node-id=20598-33756&mode=design&t=DmMCNNHxDLbpS5qq-0)

Checkout this branch `git checkout ASUB-8193_ReCaptcha_V3`
`npx fusion start -l @wpmedia/identity-block`
point the site configs on `blocks.json` to `https://subssinglesite-the-daily-intelligencer-sandbox.api.cdn.arcpublishing.com/` (It has recaptchav3 enabled for signin), or enable the config for the tenant you are using.
Create a page and includeIdentity Log In - Arc Block component. Check the component BotChallengeProtection is being rendered (reCaptcha v2 or reCaptchaV3, depending on the configs), and after ask for login the user should be logged In successfully.

## Effect Of Changes

### Before

_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After

_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[ASUB-8193]: https://arcpublishing.atlassian.net/browse/ASUB-8193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
